File: clients/java/examples/collections-insert-many-vector.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.collections.Collection;
import com.datastax.astra.client.collections.commands.results.CollectionInsertManyResult;
import com.datastax.astra.client.collections.definition.documents.Document;
import java.util.List;

public class Example {

  public static void main(String[] args) {
    // Get an existing collection
    Collection<Document> collection =
        new DataAPIClient("**APPLICATION_TOKEN**")
            .getDatabase("**API_ENDPOINT**")
            .getCollection("**COLLECTION_NAME**");

    // Insert documents to the collection
    Document document1 =
        new Document()
            .append("name", "Jane Doe")
            .append("age", 42)
            .append("$vector", new float[] {0.08f, -0.62f, 0.39f});
    Document document2 =
        new Document()
            .append("nickname", "Bobby")
            .append("$vector", new float[] {0.12f, 0.53f, 0.32f});
    CollectionInsertManyResult result = collection.insertMany(List.of(document1, document2));
    System.out.println("IDs inserted: " + result.getInsertedIds());
  }
}

File: clients/python/examples/collections-insert-many-vector.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Insert documents to the collection
# The following also demonstrates use of both plain lists and DataAPIVector
result = collection.insert_many(
    [
        {"name": "Jane Doe", "age": 42, "$vector": [0.08, -0.62, 0.39]},
        {
            "nickname": "Bobby",
            "$vector": [0.12, 0.53, 0.32],
        },
    ]
)
